### DISCUSSION TITLE
Roadmap to MVP release

### DISCUSSION THREAD

**Discussion by brafdlog** [brafdlog/caspion#144] [General]
We want to release the app to end users. Here we will discuss what is the best way to do this.

Here are my thoughts after discussing with @baruchiro:

### Release goals

- Get 30 early adopting users to install and use the app. 
- Users should be able to setup and run the app **themselves** - reach a stage where their transactions are imported and exported correctly.
- Define a clear way of sending feedback and error reports to us

### Notes
We want this release to happen soon so we won't try to make everything perfect. We should strike a balance between delivery time and the user experience.

### Milestones

0. **Merge the unify repos branch** (after open prs are merged) #51 #101 
1. **App is functional for an end user:**
    Goal: Make the basic functionality (configuration, scraping and exporting) usable for end users. UX can be terrible, but working.
    -  Choose a name and update the name in codebase #140 
    - Verify app is installable on windows #131 
    - Update the readme so it describes how to setup of the app from scratch step by step #85 
    - Define how users send logs to us in case of an error #141
    - Finish google sheets integration #120 
2. **Send app to a few friends and get feedback**
    Goal: get 5 users to run the app and give feedback
3. **Fix the most significant UX issues**
    - Fix important user feedback
    - Implement a clear first time flow (flow when there is no config yet) #142
    - Make the config ui more friendly #143
4. **Publish the app**
    - Create a landing page with basic explantation and link to download
    - Create a video showing how the app works #44 
    - Publish on social media
    - Add analytics tracking to the app #145
5. **Future**
    - Auto update flow #125 
    - Get more contributors and improve developer experience
    - Integrate sentry for errors
    - Hebrew #136 
    - Logo
    - Redesign the app #102 

I created milestones for each of these. My plan is to have a single project for the whole MVP release.

Will be happy to get feedback on this

**Comment by baruchiro** [comment]
Are we want to plan goals in the automotive tests area?

**Reply by brafdlog** [reply]
We do. But I think we should focus on that after the first version of MVP. I moved #35 to the milestone of post MVP.

What do you think?

**Reply by baruchiro** [reply]
Sure, I didn't say we need full coverage. I'm talking about giving attention to the level of the tests in the roadmap, and maybe re-enable the existed pipelines that disabled for some reason.

**Comment by baruchiro** [comment]
I think we need a simple way to notify users when we want to send them a message.

You need that before you're developing an "auto-update" or something, to let them know that they need to download a new version, or that something is broken because of something else...

I think the simple way is to create a Markdown file in this repo with the message, and requesting it from the app. If the cached file is different from the latest one, show a badge to the user to open the file in Github.

Just a simple way to note the user that we want to tell him something.

**Reply by brafdlog** [reply]
Good idea. Do you think this should be part of the MVP?

**Reply by baruchiro** [reply]
I think it should be as soon as we have users that will not be active in giving feedback.

I mean, **unless** you send the beta to specific friends and family and they are not sharing the app, you have users that not contacting you, and you don't know or forget about them, so you need to alert them to keep them updated. (And I think you will not have so many users so you can give up some of them...)

My suggestion is to implement that before we are spreading our app to beta testers that we don't know personally.

Link to the issue: #153

**Comment by Charlotte-br560** [comment]
Your roadmap is well-structured with clear goals and milestones. It's great that you're prioritizing user feedback and have plans for documentation, promotion, and future improvements.

**Reply by baruchiro** [reply]
Thank you @Charlotte-br560
